fastlane_core/spec/analytics/analytics_session_spec.rb

```diff
@@ -0,0 +1,378 @@
+      allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(File.absolute_path('./fastlane/spec/fixtures/fastfiles/'))
+      ff = Fastlane::LaneManager.cruise_lane('ios', 'beta')
+      version_events = FastlaneCore.session.events.select do |event|
+        event[:primary_target][:name] == 'fastlane_version'
+      end
+      expect(version_events.count).to eq(4)
+      action_names = version_events.map { |event| event[:action][:detail] }
+      expect(action_names).to match_array([
+                                            'default_platform',
+                                            'frameit',
+                                            'team_id',
+                                            'team_id'
+                                          ])
+    end
+
+    it 'has a completion event for each action' do
+      allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(File.absolute_path('./fastlane/spec/fixtures/fastfiles/'))
+      ff = Fastlane::LaneManager.cruise_lane('ios', 'beta')
+      completion_events = FastlaneCore.session.events.select do |event|
+        event[:action][:name] == 'completed'
+      end
+      expect(completion_events.count).to eq(4)
+      action_names = completion_events.map { |event| event[:action][:detail] }
+      expect(action_names).to match_array([
+                                            'default_platform',
+                                            'frameit',
+                                            'team_id',
+                                            'team_id'
+                                          ])
+    end
+
+    it 'has a fastfile value of true for each event' do
+      allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(File.absolute_path('./fastlane/spec/fixtures/fastfiles/'))
+      ff = Fastlane::LaneManager.cruise_lane('ios', 'beta')
+      fastfile_events = FastlaneCore.session.events.select do |event|
+        event[:primary_target][:name] == 'fastfile'
+      end
+      expect(fastfile_events.count).to eq(4)
+      fastfile_values = fastfile_events.map { |event| event[:primary_target][:detail] }
+      expect(fastfile_values).to all(be == "true")
+    end
+  end
+
+  context 'opt out' do
+    it "does not post the collected data if the opt-out ENV var is set" do
+      with_env_values('FASTLANE_OPT_OUT_USAGE' => '1') do
+        FastlaneCore.reset_session
+        allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(File.absolute_path('./fastlane/spec/fixtures/fastfiles/'))
+        ff = Fastlane::LaneManager.cruise_lane('ios', 'beta')
+        completion_events = FastlaneCore.session.events.select do |event|
+          event[:action][:name] == 'completed'
+        end
+        expect(completion_events.count).to eq(4)
+        expect(FastlaneCore.session.client).not_to receive(:post_events)
+        FastlaneCore.session.finalize_session
+      end
+    end
+  end
+end
+
+# here are a bunch of tests we should also have
```

Comment:
What can we do to keep these from slipping through the cracks?  🤔

Reply:
made a trello card for this!